ext/oboe_metal/README.md

```diff
@@ -65,3 +65,86 @@ Some inspiring examples here:
 <https://jvns.ca/blog/2016/06/12/a-weird-system-call-process-vm-readv/>
 
 <https://medium.com/@zanker/finding-a-ruby-bug-with-gdb-56d6b321bc86>
+
+## Debug the c-code with core dump
+
+If the core dump is produced, it's easier to check the backtrace with it in gdb
+
+For example, `(core dumped)` indicate the crash file is dumped
+
+```console
+./start.sh: line 37:    65 Segmentation fault      (core dumped) ...
+```
+
+### Prepare
+
+#### 1. Install solarwinds_apm with debug symbol on
+
+```console
+export OBOE_DEBUG=true  # enable debug flag when compiling; and also download *.debug into lib/ when create the gem for ease of use
+export OBOE_DEV=true    # optional: if you want to install the nightly build liboboe
+
+gem install solarwinds_apm
+```
+
+#### 2. Check the core dump size is not constrained
+
+```console
+ulimit -c unlimited       # have to set this for unlimited core dump file size without trimmed error message
+```
+
+#### 3. Check if the crash report program is configured correctly
+
+Ubuntu use [apport](https://wiki.ubuntu.com/Apport); Debian use [kdump](https://mudongliang.github.io/2018/07/02/debian-enable-kernel-dump.html)
+
+In ubuntu, if disable the apport by `service apport stop`, the core dump file will be stored in the current directory and named as `core`
+
+#### 4. Gather the `*.debug` file for oboe symbol
```

Comment:
Presumably the liboboe debug symbols file from the agent-binaries fileserver, yes?

This is great info, but we had a discussion in slack (just tagged you) where @jerrytfleung said his plan is to include the debug symbols for the "RelWithDebInfo" build--so maybe we can just go with using that version of liboboe instead.

Reply:
Will update it once jerry has that in place